src/plugins/explore/public/components/visualizations/gauge/to_expression.ts

```diff
@@ -29,19 +30,27 @@ export const createGauge = (
   let numericalValues: number[] = [];
   let maxNumber: number = 0;
   if (numericField) {
-    numericalValues = transformedData
-      .map((d) => Number(d[numericField]))
-      .filter((n) => !Number.isNaN(n));
-
+    numericalValues = transformedData.map((d) => d[numericField]);
     maxNumber = Math.max(...numericalValues);
   }
 
   const styleOptions = { ...defaultGaugeChartStyles, ...styles };
 
   const calculatedValue = calculateValue(numericalValues, styleOptions.valueCalculation);
 
+  const isValidNumber =
+    calculatedValue !== undefined && typeof calculatedValue === 'number' && !isNaN(calculatedValue);
+
   const targetValue = calculatedValue || 0;
 
+  const selectedUnit = getUnitById(styleOptions?.unitId);
+
+  const displayValue = isValidNumber
+    ? selectedUnit && selectedUnit?.display
+      ? selectedUnit?.display(calculatedValue, selectedUnit?.symbol)
+      : `${Math.round(calculatedValue * 100) / 100} ${selectedUnit?.symbol ?? ''}`
+    : '-';
```

Comment:
Can we refactor it for better readability? And maybe extract it to util function, I see it's repeating